src/LBFactory.sol

```diff
@@ -357,17 +352,35 @@ contract LBFactory is Ownable2Step, AccessControl, ILBFactory {
         }
 
         {
-            address implementation = _lbPairImplementation;
+            bytes memory data = abi.encodeWithSignature("initialize(uint16,uint16,uint16,uint16,uint24,uint16,uint24,uint24)",
+                preset.getBaseFactor(),
+                preset.getFilterPeriod(),
+                preset.getDecayPeriod(),
+                preset.getReductionFactor(),
+                preset.getVariableFeeControl(),
+                preset.getProtocolShare(),
+                preset.getMaxVolatilityAccumulator(),
+                activeId
+            );
 
-            if (implementation == address(0)) revert LBFactory__ImplementationNotSet();
+            bytes32 salt = keccak256(abi.encodePacked(tokenA, tokenB, binStep));
 
-            pair = ILBPair(
-                ImmutableClone.cloneDeterministic(
-                    implementation,
-                    abi.encodePacked(tokenX, tokenY, binStep),
-                    keccak256(abi.encode(tokenA, tokenB, binStep))
-                )
+            bytes memory bytecode = abi.encodePacked(
+                type(LBPairBeaconProxy).creationCode,
+                abi.encode(_lbPairBeacon, tokenX, tokenY, binStep, data)
             );
+
+             // Deterministic deployment
+            address pairAddress;
+            assembly {
+                pairAddress := create2(0, add(bytecode, 0x20), mload(bytecode), salt)
+                if iszero(extcodesize(pairAddress)) {
+                    revert(0, 0)
```

Comment:
this revert is more informative: https://github.com/OpenZeppelin/openzeppelin-contracts/blob/master/contracts/utils/Create2.sol#L57
btw why not just use the OZ create2 library for deterministic deployment?

Reply:
Updated in this [commit](https://github.com/DistributedCollective/sovryn-lb-dex/pull/3/commits/33e4869315cfab629c563797f7a0dd75076205b6)